connectors/src/lib/data_sources.ts

```diff
@@ -1124,3 +1125,130 @@ export async function deleteTable({
     throw new Error(`Error deleting from dust: ${dustRequestResult}`);
   }
 }
+
+export async function upsertFolderNode({
```

Comment:
nit: we could have started to use dustApi (like in `tokenize` from the same file)
leaving you to decide

Reply:
definitely, I'll replace that !

Reply:
✅